dm/checker/cmd.go

```diff
@@ -53,25 +56,29 @@ func CheckSyncConfig(ctx context.Context, cfgs []*config.SubTaskConfig, errCnt,
 	}
 	checkingItems := config.FilterCheckingItems(ignoreCheckingItems)
 	if len(checkingItems) == 0 {
-		return nil
+		return "", nil
 	}
 
 	c := NewChecker(cfgs, checkingItems, errCnt, warnCnt)
 
 	if err := c.Init(ctx); err != nil {
-		return terror.Annotate(err, "fail to initial checker")
+		return "", terror.Annotate(err, "fail to initial checker")
 	}
 	defer c.Close()
 
 	pr := make(chan pb.ProcessResult, 1)
 	c.Process(ctx, pr)
-	for len(pr) > 0 {
+	if len(pr) > 0 {
 		r := <-pr
```

Comment:
how about we always assume there is one result in `pr`, as stated in https://github.com/pingcap/tiflow/blob/d8dab8df9663b831d9ad6556ce48a9cfd0a9b770/dm/dm/unit/unit.go#L44-L47

Reply:
It can be seen that `checker` is designed according to the unit at the beginning. But it is not in fact(not complete unit interface). And I think we don't plan to treat him as a unit in the future yet. So it is a semi-finished product. This is the reason that it looks a lot unreasonable. 

I think we can modify these unreasonable places after the end of this feature. Result is also. @Ehco1996

Reply:
actully it depends on requirements, i am not sure if we have a plan to merge full-time/sync-time sync_diff in to checker? if so i thinks it's better to treat checker as a unit